README.md

```diff
@@ -12,16 +12,96 @@ First generate your definition with the builder.
 The definition defines the schemas and collections you wanna use.
 
 ```js
-// TODO (see ./example here for now)
+// build.js
+const Hyperschema = require('hyperschema')
+const HyperDB = require('hyperdb/builder')
+
+const SCHEMA_DIR = './spec/hyperschema' // where the schema definitions are written
+const DB_DIR = './spec/hyperdb' // Where to install db definition
+
+// Hyperschema definitions
+const schema = Hyperschema.from(SCHEMA_DIR)
+const example = schema.namespace('example')
+
+// Define 'members'
+example.register({
+  name: 'members',
+  compact: true,
+  fields: [
+    {
+      name: 'name',
+      type: 'string',
+      required: true,
+    },
+    {
+      name: 'age',
+      type: 'uint',
+      required: true,
+    },
+  ]
+})
+Hyperschema.toDisk(schema)
+
+// Hyperdb collection definitions
+const db = HyperDB.from(SCHEMA_DIR, DB_DIR)
+const exampleDB = db.namespace('example')
+
+// Import helpers (see next step)
+exampleDB.require('./helpers.js')
+
+// Define collections of structs
+exampleDB.collections.register({
+  name: 'members',
+  schema: '@example/members',
+  key: ['name']
+})
+
+// Define index over collection
+exampleDB.indexes.register({
+  name: 'members-by-name',
+  collection: '@example/members',
+  unique: true,
+  key: {
+    type: 'string',
+    map: 'mapNameToLowerCase',
+  }
+})
+
+HyperDB.toDisk(db)
 ```
 
+Define helper functions in `helper.js`:
+
+```js
+// helper.js
+exports.mapNameToLowerCase = (record, context) => {
+  const name = record.name.toLowerCase().trim()
+  return name ? [name] : []
+}
+```
+
+Run `node build.js` to build the spec.
+
 Then boot your db. You can use the same definition for a fully local db and a P2P one.
 
 ``` js
+// run.js
 const HyperDB = require('hyperdb')
 
-// first choose your engine
-const db = HyperDB.rocks('./my-rocks.db', require('./my-definition'))
+;(async () => {
```

Comment:
fine to just remove the async wrapper here, i normally ignore that part for simplicity (feel free to esm import above)

Reply:
Done [ce0f6da](https://github.com/holepunchto/hyperdb/pull/43/commits/ce0f6da3a130ea16f0789477655fbbb67b0aa65f)!